### PR TITLE
wrangler: some additions

### PR DESCRIPTION
**Pull request by elithrar**
This PR adds:

- a new command surface: `wrangler vectorize`
- support for the new `[[vectorize]]` binding

```
➜  node bin/wrangler.js vectorize --help
wrangler vectorize

🧮 Interact with Vectorize indexes

Commands:
  wrangler vectorize create <name>  Create a Vectorize index
  wrangler vectorize delete <name>  Delete a Vectorize index
  wrangler vectorize get <name>     Get a Vectorize index by name
  wrangler vectorize list           List your Vectorize indexes
```

**Author has included the following, where applicable:**

- [x] Tests
- [x] Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))

### CODE REVIEW

**Comment by elithrar** [comment]
I've removed the 'bad' mTLS test, which a) had the wrong snapshot and b) shouldn't force other commands to update the mTLS test when they add a new top-level command.

File: packages/wrangler/src/__tests__/mtls-certificates.test.ts
Comment (penalosa) [inline packages/wrangler/src/__tests__/mtls-certificates.test.ts:357]: Instead of removing this test, could you fix the typo so it's testing what the test name describes?

File: packages/wrangler/src/__tests__/vectorize/vectorize.test.ts
Comment (penalosa) [inline packages/wrangler/src/__tests__/vectorize/vectorize.test.ts:6]: It would be great to see more tests here—currently it's just testing the help output, and there's no testing relating to `vectorize` functionality

File: packages/wrangler/src/index.ts
Comment (penalosa) [inline packages/wrangler/src/index.ts:8]: I don't think this PR should change anything `ai` related?

File: packages/wrangler/src/index.ts
Comment (penalosa) [inline packages/wrangler/src/index.ts:459]: This looks like maybe some merge conflicts snuck into the PR: this shouldn't remove any `ai`-related functionality

File: packages/wrangler/src/index.ts
Comment (penalosa) [inline packages/wrangler/src/index.ts:471]: This is my britishness talking, but I'd _love_ to see a hidden `vectorise` alias for this command 😅 (non-blocking, probably not a good idea)

File: packages/wrangler/src/vectorize/common.ts
Comment (penalosa) [inline packages/wrangler/src/vectorize/common.ts:2]:
```suggestion
	`--------------------
📣 Vectorize is currently in open beta
📣 See the Vectorize docs for how to get started and known issues: https://developers.cloudflare.com/vectorize
📣 Please report any bugs to https://github.com/cloudflare/workers-sdk/issues/new/choose
📣 To give feedback, visit https://discord.cloudflare.com/
--------------------
`;
```

File: .changeset/good-ads-fold.md
Comment (penalosa) [inline .changeset/good-ads-fold.md:5]: Ideally this would have more details (example commands etc...), but maybe a link to the docs would work?

File: packages/wrangler/src/vectorize/index.ts
Comment (penalosa) [inline packages/wrangler/src/vectorize/index.ts:29]: Should this be `indices`?

File: packages/wrangler/src/vectorize/insert.ts
Comment (penalosa) [inline packages/wrangler/src/vectorize/insert.ts:23]:
```suggestion
					"An array of one or more vectors in JSON format to insert into the index",
```

File: packages/wrangler/src/vectorize/insert.ts
Comment (penalosa) [inline packages/wrangler/src/vectorize/insert.ts:38]: Assuming this is WIP?

File: packages/wrangler/src/vectorize/list.ts
Comment (penalosa) [inline packages/wrangler/src/vectorize/list.ts:30]:
```suggestion
		logger.warn(`
You haven't created any indexes on this account.
```

File: packages/wrangler/src/vectorize/list.ts
Comment (penalosa) [inline packages/wrangler/src/vectorize/list.ts:48]:
```suggestion
			description: index.description ?? "",
```

File: packages/wrangler/src/vectorize/query.ts
Comment (penalosa) [inline packages/wrangler/src/vectorize/query.ts:23]:
```suggestion
					"The vector value to query for",
```

File: packages/wrangler/src/vectorize/query.ts
Comment (penalosa) [inline packages/wrangler/src/vectorize/query.ts:27]:
```suggestion
			"top-k": {
```

File: packages/wrangler/src/vectorize/delete.ts
Comment (penalosa) [inline packages/wrangler/src/vectorize/delete.ts:23]: We call this type of thing `--force` in a couple other places, so it should probably be the same here for consistency

File: packages/wrangler/src/vectorize/delete.ts
Comment (penalosa) [inline packages/wrangler/src/vectorize/delete.ts:35]:
```suggestion
			`OK to delete the index '${args.name}'?`
```

**Review by penalosa — COMMENTED** [review]
General comment around renaming from `indexes`-> `indices`, but other than that this is looking pretty good! Couple merge issues, and it looks like some bits are still WIP, but no major issues at this stage.

File: .changeset/good-ads-fold.md
Comment (elithrar) [inline .changeset/good-ads-fold.md:5]:
```suggestion
Adds wrangler support for Vectorize, Cloudflare's new vector database, under `wrangler vectorize`. Visit the developer
documentation (https://developers.cloudflare.com/vectorize/) 
to learn more and create your first vector database with
`wrangler vectorize create my-first-index`.
```

Comment (elithrar) [reply .changeset/good-ads-fold.md:5]: Good call. Fixing.

File: packages/wrangler/src/vectorize/insert.ts
Comment (elithrar) [reply packages/wrangler/src/vectorize/insert.ts:38]: Yep, both `query` and `insert` are stubbed out and will be fast follows.

**Comment by elithrar** [comment]
@penalosa this should be OK to merge from my perspective, pending any final review.

Both `vectorize query` and `vectorize insert` are stubbed out, and we'll be following up with these in a few days, but don't want to miss the next release.

**Comment by elithrar** [comment]
Waiting on final review & merge.

**Comment by elithrar** [comment]
> General comment around renaming from `indexes`-> `indices`, but other than that this is looking pretty good! Couple merge issues, and it looks like some bits are still WIP, but no major issues at this stage.

We’re going to stick with “indexes”, since this is convention in this use-case and not incorrect (e.g. database indexes, not indices).

File: packages/wrangler/src/cfetch/internal.ts
Comment (elithrar) [reply packages/wrangler/src/cfetch/internal.ts:93]: Right now, wrangler does not handle the HTTP 204 response (which is semantically correct for a DELETE) from our API. The internal fetch wrapper _always_ expects a HTTP response body.

We either:

1. Change our API (not going to happen right now)
2. Let users get an ugly error that makes it unclear if their `delete <index-name>` worked
3. Patch this.

I think 3 is the right choice here.

**Comment by penalosa** [comment]
@elithrar Mostly looking good, but it would be great to see some tests of the added Wrangler functionality, including:
- tests of the configuration validation
- tests of the new command surface, with mocked out implementations of the underlying APIs

File: packages/wrangler/src/__tests__/vectorize/vectorize.test.ts
Comment (elithrar) [reply packages/wrangler/src/__tests__/vectorize/vectorize.test.ts]: addressed.

File: packages/wrangler/src/vectorize/create.tsx
Comment (elithrar) [reply packages/wrangler/src/vectorize/create.tsx:122]: Is this blocking? We have existing code using `renderToString` here.

File: packages/wrangler/src/cfetch/internal.ts
Comment (penalosa) [reply packages/wrangler/src/cfetch/internal.ts:93]: Can you add a comment to that effect? I'm mostly concerned about the impact on other service's API endpoints. Long-term, I think the right solution here is to check for the presence of a body, and return `null` if non exists

**Review by JacobMGEvans — APPROVED** [review]
Ran Prettier. Looks good, thanks for adding tests!!

File: packages/wrangler/src/cfetch/internal.ts
Comment (elithrar) [reply packages/wrangler/src/cfetch/internal.ts:95]: Done

**Review by penalosa — APPROVED** [review]

**Pull request merged**
